library/public/interactor.h

```diff
@@ -65,77 +116,89 @@ class F3D_EXPORT interactor
    *
-   * interaction can be a pressed key symbol, eg: "C",
+   * Bind modifiers is a binary flag from the dedicated enum that represent KeyModifiers.
+   * Bind interaction can be a pressed key symbol, eg: "C",
    * or a dedicated key symbol for special keys:
    * "Left", "Right", "Up", "Down", "Space", "Enter", "Escape", "Question".
    *
-   * modifiers is a binary flag from the dedicated enum that represent KeyModifiers.
+   * group is an optional arg to group bindings together for better display of the documentation.
+   * Groups are kept in order of addition when recovered using `getBindGroups`.
+   * Bindings are kept in order of addition when recovered using `getBindsForGroup`.
    *
-   * When the corresponding interaction and modifiers happens, the provided commands will be
-   * triggered using triggerCommand.
-   * Considering checking if an interaction exists or removing it before adding it to avoid
-   * potential conflicts.
+   * documentationCallback is an optional function that returns a pair of string,
+   * the first is the doc itself, the second is the current value as a string, if any.
+   * Use `getBindingDocumentation` to access this doc.
+   *
+   * When the corresponding bind happens, the provided commands will be triggered using
+   * triggerCommand. Considering checking if an interaction exists or removing it before adding it
+   * to avoid potential conflicts.
    *
    * ANY modifier interactions will only be triggered if no other interaction bind with modifier
    * is found.
    *
-   * Adding commands for an existing combination of interaction and modifier will throw a
-   * interactor::already_exists_exception.
+   * Adding commands for an existing bind will throw a interactor::already_exists_exception.
    */
-  virtual interactor& addBinding(
-    const std::string& interaction, ModifierKeys modifiers, std::vector<std::string> commands) = 0;
+  virtual interactor& addBinding(const interaction_bind_t& bind, std::vector<std::string> commands,
+    std::string group = std::string(),
+    std::function<std::pair<std::string, std::string>()> documentationCallback = nullptr) = 0;
 
   /**
    * See addBinding
-   * Convenience method to add a single command for an interaction, similar as
-   * addBinding(interaction, modifiers, {command})
+   * Convenience method to add a single command for an interaction,
+   * similar as `addBinding(bind, {command})`
    *
-   * Adding command for an existing combination of interaction and modifier will throw a
-   * interactor::already_exists_exception.
+   * Adding command for an existing bind will throw a interactor::already_exists_exception.
    */
-  virtual interactor& addBinding(
-    const std::string& interaction, ModifierKeys modifiers, std::string command) = 0;
+  virtual interactor& addBinding(const interaction_bind_t& bind, std::string command,
+    std::string group = std::string(),
```

Comment:
```suggestion
    std::string group = {},
```
or
```suggestion
    std::string group = "",
```
to avoid repetition of string keyword?